### PR TITLE
Min Initial Deposit (Recreated using main as base branch)

### PR DESCRIPTION
**Pull request by fragwuerdig**
## Summary of changes

See #74 and #41.

## Report of required housekeeping

- [ ] Github issue OR spec proposal link
- [ ] Wrote tests
- [ ] Updated API documentation (client/lcd/swagger-ui/swagger.yaml)
- [ ] Added a relevant changelog entry: clog add [section] [stanza] [message]

----

## (FOR ADMIN) Before merging

- [ ] Added appropriate labels to PR
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" (coding standards)
- [ ] Confirm added tests are consistent with the intended behavior of changes
- [ ] Ensure all tests pass

### CODE REVIEW

**Comment by fragwuerdig** [comment]
@nghuyenthevinh2000: I recreated this PR (now based non main). Can you test the upgrade handler/migration for this? I fear the automated upgrade test is really doing nothing... I would appreciate you manual test that you have been doing for #74.

**Comment by fragwuerdig** [comment]
FYI: I still have to add the unit-tests.

File: app/app.go [hunk 1]
```diff
@@ -569,6 +569,7 @@ func NewTerraApp(
 			SignModeHandler:    encodingConfig.TxConfig.SignModeHandler(),
 			IBCChannelKeeper:   app.IBCKeeper.ChannelKeeper,
 			DistributionKeeper: app.DistrKeeper,
+			GovKeeper:			app.GovKeeper,
```
Comment (ZaradarBH) [inline app/app.go:572]: Lint'ers might complain about the extra indentation in this line :)

File: custom/auth/ante/ante.go [hunk 2]
```diff
@@ -48,6 +49,10 @@ func NewAnteHandler(options HandlerOptions) (sdk.AnteHandler, error) {
 		return nil, sdkerrors.Wrap(sdkerrors.ErrLogic, "sign mode handler is required for ante builder")
 	}
 
+	if options.GovKeeper == nil {
```
Comment (ZaradarBH) [inline custom/auth/ante/ante.go:52]: Maybe we should add similar checks for the DistributionKeeper &  IBCChannelKeeper?

File: custom/auth/ante/min_initial_deposit.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
+package ante
+
+import (
+	"fmt"
+	//"strconv"
```
Comment (ZaradarBH) [inline custom/auth/ante/min_initial_deposit.go:5]: Might as well just remove this import if its not being used :)

**Review by ZaradarBH — COMMENTED** [review]
LGTM m8. Only have a few minor  cosmetic suggestions :)

**Review by nghuyenthevinh2000 — APPROVED** [review]

**Review by nghuyenthevinh2000 — CHANGES_REQUESTED** [review]
Base logic looks fine
1. a test should be needed 
2. func (AppModule) ConsensusVersion() uint64 { return 2 } in module.go should return 3

File: custom/auth/ante/ante.go (hunk 2 above)
Comment (fragwuerdig) [reply custom/auth/ante/ante.go:52]: this does not belong to the scope of this PR... I will make another PR for this. Can be resolved?

File: custom/auth/ante/min_initial_deposit.go (hunk 3 above)
Comment (fragwuerdig) [reply custom/auth/ante/min_initial_deposit.go:5]: I actually need that import. So can we keep it?

Comment (fragwuerdig) [reply custom/auth/ante/min_initial_deposit.go:5]: image

**Comment by fragwuerdig** [comment]
> Base logic looks fine
> 
> 1. a test should be needed
> 2. func (AppModule) ConsensusVersion() uint64 { return 2 } in x/treasury/module.go should return 3
> 3. you need to add an upgrade handler in app/upgrades as well since this is state migration in x/treasury. It is also prerequisite to run upgrade-test ci

Ok. Added unit test and also made that upgrade magic happen.

File: custom/auth/ante/min_initial_deposit.go (hunk 3 above)
Comment (ZaradarBH) [reply custom/auth/ante/min_initial_deposit.go:5]: I was thinking about "strconv" which is commented out at the moment :)

**Comment by fragwuerdig** [comment]
i wrecked this branch *argh* - that's because I do things in a hurry........

**Comment by ZaradarBH** [comment]
> i wrecked this branch _argh_ - that's because I do things in a hurry........

image

File: custom/auth/ante/min_initial_deposit.go (hunk 3 above)
Comment (fragwuerdig) [reply custom/auth/ante/min_initial_deposit.go]: ok - removed it

**Comment by fragwuerdig** [comment]
Ok. I saved the branch 🤣. Wrote another test for this (half-automated spinning single validator node doing the min initial deposit prop via CLI). Where do I put this script? @nghuyenthevinh2000

**Comment by ZaradarBH** [comment]
> Ok. I saved the branch 🤣. Wrote another test for this (half-automated spinning single validator node doing the min initial deposit prop via CLI). Where do I put this script?
> 
> @nghuyenthevinh2000 : I also comply with your change requests. Can you resolve them?

Normally ad-hoc scripts would go into the "contrib" folder, but I recon we can drop it in the new "tests" folder  we are setting up for the automated testing stuff :)

**Review by ZaradarBH — APPROVED** [review]

**Review by nghuyenthevinh2000 — APPROVED** [review]
LGTM

**Pull request merged**